File: Php/app/User/UserNavigationModel.php
namespace app\User{

	use core\Navigation\NavigationModel;
	use app\Urls\AuthLogoutUrlModel;
	use app\Urls\SharingSettingsUrlModel;
	use app\Urls\UserDetailsUrlModel;
	use Domain\Classes\User;
	use core\Navigation\NavigationNode;
	use app\Urls\ForgotPasswordUrlModel;
	use app\Urls\UserAddUrlModel;
	use app\Urls\AuthLoginUrlModel;

	class UserNavigationModel extends NavigationModel{

		public function __construct(User $user = null){
			heading = 'Account';
			cssClass = 'user-nav';

			if($user == null){
				setupAnonymous();
			} else {
				setupLoggedIn($user);
			}
		}

		private function setupAnonymous(){
			addNode(new NavigationNode('Sign in', new AuthLoginUrlModel()));
			addNode(new NavigationNode('Register', new UserAddUrlModel()));
			addNode(new NavigationNode('Forgot password', new ForgotPasswordUrlModel()));
		}

		private function setupLoggedIn(User $user){
			addNode(new NavigationNode($user.getDisplayName(), new UserDetailsUrlModel($user)));
			addNode(new NavigationNode('Sharing', new SharingSettingsUrlModel()));
			addNode(new NavigationNode('Sign Out', new AuthLogoutUrlModel()));
		}

	}

}

File: Web/Models/NavigationModels/UserNavigationModel.cs
using System.Collections.Generic;
using Core.UseCases;
using PokerBunch.Common.Urls.SiteUrls;

namespace Web.Models.NavigationModels
{
    public class UserNavigationModel : NavigationModel
    {
        public UserNavigationModel(CoreContext.Result appContextResult)
        {
            Heading = "Account";
            CssClass = "user-nav";
            Nodes = GetNodes(appContextResult);
        }

        private IList<NavigationNode> GetNodes(CoreContext.Result appContextResult)
        {
            return appContextResult.IsLoggedIn ? GetLoggedInNodes(appContextResult) : GetAnonymousNodes();
        }

        private IList<NavigationNode> GetAnonymousNodes()
        {
            return new List<NavigationNode>
                {
                    new NavigationNode("Sign in", new LoginUrl().Relative),
                    new NavigationNode("Register", new AddUserUrl().Relative),
                    new NavigationNode("Reset password", new ResetPasswordUrl().Relative)
                };
        }

        private IList<NavigationNode> GetLoggedInNodes(CoreContext.Result appContextResult)
        {
            return new List<NavigationNode>
                {
                    new NavigationNode($"Signed in as {appContextResult.UserDisplayName}", new UserDetailsUrl(appContextResult.UserName).Relative),
                    new NavigationNode("Sign Out", new LogoutUrl().Relative)
                };
        }
    }
}